### PR TITLE
Upgrade to styled-components v6

### PR DESCRIPTION
**Pull request by goplayoutside3**
## Package
app-root
app-project
lib-classifier
lib-content
lib-react-components
lib-subject-viewers
lib-user

## Describe your changes

- Bump all apps and libraries to latest version of styled-components.
- Standardize use of `babel-plugin-styled-components`.
- Modify spec files for components with `useTheme()` that previously didn't have a ThemeProvider wrapper.
- Some minor changes following the migration guide.
- Delete `ignore-styles` third party from the whole monorepo. It's unmaintained and not useful anymore.

## How to Review

I did the following UI checks and it would be helpful to double check a few of them during review. See more comments in the diff view about specific changes.
- [ ] i-fancy-cats does not generate new console warnings about styling
- [ ] i-fancy-cats UI does not look any different in all major browsers
- [ ] All Storybooks in FEM build and run
- [ ] All of app-root pages' UI is unchanged, and no new console warnings
- [ ] All unit tests pass

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [ ] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [ ] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [ ] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

## Maintenance
- [ ] If not from dependabot, the PR creator has described the update (major, minor, or patch version, changelog)

### CODE REVIEW

File: packages/app-project/.babelrc [hunk 1]
```diff
@@ -2,6 +2,7 @@
   "env": {
     "test": {
       "plugins": [
+        "babel-plugin-styled-components",
```
Comment (goplayoutside3) [inline packages/app-project/.babelrc:5]: This plugin was previously used only in lib-content and lib-react-components. The plugin docs are [here](https://styled-components.com/docs/tooling#babel-plugin), but it helps with creating human-readable classnames for components, server side rendering, and transpiling template literals. I added it to all apps and libraries in FEM.

File: packages/app-project/src/helpers/theme/README.md [hunk 2]
```diff
@@ -1,5 +1,3 @@
 # Theme
 
 Creates a Grommet theme by performing a deep merge of the Grommet base theme with the Zooniverse custom Grommet theme.
-
-This was originally done in the `GrommetWrapper` theme, but was moved here so it can be reused with `@artsy/fresnel` to extract the breakpoints for responsive sizing.
```
Comment (goplayoutside3) [inline packages/app-project/src/helpers/theme/README.md:5]: `@artsy/fresnel` was removed awhile ago. This helper could probably be removed, but will do so in different PR.

File: packages/app-project/src/screens/ProjectHomePage/components/Hero/components/Background/Background.js [hunk 3]
```diff
@@ -1,45 +1,30 @@
...
-      flex: 1 1 auto;
-      object-position: 0 50%;
-    }
-  `}
+  // Grommet small breakpoint
+  @media (width > 769px) {
+    flex: 1 1 auto;
+    object-position: 0 50%;
+  }
 `
```
Comment (goplayoutside3) [inline packages/app-project/src/screens/ProjectHomePage/components/Hero/components/Background/Background.js:15]: Passing props like `breakpoint` throws a warning in v6 because [shouldForwardProp is no longer the default](https://styled-components.com/docs/faqs#shouldforwardprop-is-no-longer-provided-by-default). I refactored this Background component to simply use CSS breakpoints of a defined pixel unit because that's what we do in other places like lib-content.

File: packages/app-root/src/app/style-registry.js [hunk 4]
```diff
@@ -22,7 +22,7 @@ export default function StyledComponentsRegistry({
   if (typeof window !== 'undefined') return <>{children}</>
 
   return (
-    <StyleSheetManager disableVendorPrefixes sheet={styledComponentsStyleSheet.instance}>
+    <StyleSheetManager sheet={styledComponentsStyleSheet.instance}>
```
Comment (goplayoutside3) [inline packages/app-root/src/app/style-registry.js:25]: `disableVendorPrefixes` is no longer recommended per the migration guide.

File: packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SingleImageViewer/SingleImageViewer.js [hunk 5]
```diff
@@ -11,8 +11,8 @@ import locationValidator from '../../helpers/locationValidator'
 const PlaceholderSVG = styled.svg`
   background: no-repeat center / cover url('https://static.zooniverse.org/www.zooniverse.org/assets/fe-project-subject-placeholder-800x600.png');
   touch-action: pinch-zoom;
-  max-width: ${props => props.maxWidth || '100%'};
-  ${props => props.maxHeight && css`max-height: ${props.maxHeight};`}
+  max-width: ${props => props.$maxWidth || '100%'};
+  ${props => props.$maxHeight && css`max-height: ${props.$maxHeight};`}
```
Comment (goplayoutside3) [inline packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SingleImageViewer/SingleImageViewer.js:15]: The addition of `$` is a [transient props](https://styled-components.com/docs/faqs#transient-props-since-5.1) strategy to avoid the warning generated by passing non-valid HTML attributes to the element. We rarely do this in FEM, so this change is unique to SingleImageViewer in this PR.

File: packages/lib-react-components/src/Markdownz/Markdownz.stories.js [hunk 6]
```diff
@@ -1,17 +1,11 @@
...
 import Markdownz from './Markdownz'
 import readme from './README.md'
 import markdownExample from '../../.storybook/lib/example.md'
 import { examples } from './helpers/storybookExamples'
 import markdownInGrid from './markdownGridExample.md'
 
-const TableRowWithBorder = styled(TableRow)`
-  border-top: solid thin black;
-  border-bottom: solid thin black;
-`
```
Comment (goplayoutside3) [inline packages/lib-react-components/src/Markdownz/Markdownz.stories.js:13]: This is the only case where `styled()` was used in FEM in a dev environment. There's no reason for libraries in FEM to install `styled-components` as a devDep so I refactored this Story with a different, but similar "with custom-styled components" test case.

**Review by kieftrav — APPROVED** [review]
LGTM - I didn't notice any UI changes. I love seeing the tech debt cleanup!
